### PR TITLE
Added documentation for LORE release workflow

### PR DESCRIPTION
**Pull request by pwilkins**
Creating LORE release is a multi-step, manual process. This document
describes that process so that it can be done by any team member.
- Added `release_process.rst` file.
- Corrected formatting in `util/release_notes_rst.ejs`.
- Added `release_process.rst` file to Sphinx build.
- Fixed display of version number in Bootstrap header.

@pdpinch

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #pylint: skip-file
 
 sys.path.insert(0, os.path.abspath('..'))
+from lore.settings import VERSION
```
Comment (pdpinch) [inline docs/conf.py:9]: Isn't this what got us in trouble with readthedocs?

File: docs/release_process.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+.. code-block:: bash
+
+    git fetch
+    git checkout master
+    git pull
+    git flow release start 0.10.0
+    vi lore/settings.py # change version number
+    vi util/release_notes_rst.ejs # change version temporarily -- don't commit!
+    # find the hashes for the first and last commits to include
+    git-release-notes v0.9.0..master util/release_notes_rst.ejs
```
Comment (pdpinch) [inline docs/release_process.rst]: Does git-release-notes work with tags? I used it with hashes before. For example if first commit is `23b195` and last commit is `7c84a4`

`git-release-notes 23b195..7c84a4 util/release_notes_rst.ejs`

File: docs/release_process.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+- Update ``release.rst`` with this version's release notes (the file
+  generated by running ``release_notes_rst.ejs``).
+
+4. Create Pull Requests
+-----------------------
+
+Push the rc branch and make three PRs:
+  - one to ``master`` called ``Release 0.10.0``
+  - one to ``release`` called ``rc/0.10.0`` containing a checklist of commits
```
Comment (pdpinch) [inline docs/release_process.rst]: I believe `git flow release publish 0.10.0` will do these two steps (but not the next one).

File: docs/release_process.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  git-release-notes v0.9.0..master util/release_notes.ejs
+
+and paste the Markdown output as the description of the PR to the release
+branch ``rc/0.10.0`` (This is the one with the checkboxes.)
+
+Tag the one to master to be reviewed ``Release 0.10.0``
+
+.. code-block:: bash
+
+  ``git tag Release 0.10.0``
```
Comment (pdpinch) [inline docs/release_process.rst]: I believe `git flow release finish 0.10.0` will take care of creating the tag.

File: docs/release_process.rst [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+=============
+
+Don't hesitate to merge to the ``release-candidate`` branch, because
+folks will not wait for it to show up to start their verification tests.
+In retrospect, I probably should have done this *before* creating the PR
+to the ``release`` branch with all the checkboxes. The one to the
+``release-candidate`` branch should be merged immediately to kick
+of the build.
+
+``git flow release finish 0.10.0``
```
Comment (pdpinch) [inline docs/release_process.rst]: We should add steps throughout about confirming that the builds were successful.

File: docs/release_process.rst (hunk 2 above)
Comment (pdpinch) [inline docs/release_process.rst:23]: Cool. I didn't realize this works with tags and branches

File: docs/conf.py (hunk 1 above)
Comment (pwilkins) [inline docs/conf.py:9]: Something like it, but this time rtd is happy with the result.  See http://lore.readthedocs.org/en/feature-pw-add_release_process_notes/

**Comment by pdpinch** [comment]
You need a step that corresponds to 

```
vi lore/settings.py # change version number
vi util/release_notes_rst.ejs # change version temporarily -- don't commit!
```

File: docs/release_process.rst [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+``git flow release start 0.10.0``
+
+3. Generate the Release Notes
+-----------------------------
+Create the release notes using a utility called ``git-release-notes``.
+https://www.npmjs.com/package/git-release-notes
+
+- Update the ``VERSION`` number in ``lore/settings.py`` following Semantic
+  numbering practice.
```
Comment (pdpinch) [inline docs/release_process.rst]: I think updating the version number should be a separate step from generating the release notes.

File: docs/release_process.rst [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+.. code-block:: bash
+
+  git-release-notes v0.9.0..master util/release_notes_rst.ejs
+
+- Update ``release.rst`` with this version's release notes (the file
+  generated by running ``release_notes_rst.ejs``).  Edit the new
+  release notes to make each statement in the past tense, capitalize
+  the first word and end sentences with a period.
+
```
Comment (pdpinch) [inline docs/release_process.rst:77]: I think this where you need the commit, after the version number has been updated and after release.rst has been updated.

File: docs/release_process.rst [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  git-release-notes v0.9.0..master util/release_notes.ejs
+
+and paste the Markdown output as the description of the PR to the release
+branch ``rc/0.10.0`` (This is the one with the checkboxes.)
+
+Push the rc branch and make three PRs:
+  - one to ``master`` called ``Release 0.10.0``
+  - one to ``release`` called ``rc/0.10.0`` containing a checklist of commits
+  - one to ``release-candidate`` called ``rc/0.10.0`` which should be merged
+    promptly to kick of the build
```
Comment (pdpinch) [inline docs/release_process.rst]: This PR (to release candidate) should be done 1st, so you can check to see if the build is successful.

File: docs/release_process.rst (hunk 6 above)
Comment (pwilkins) [inline docs/release_process.rst]: Agreed.

File: docs/release_process.rst (hunk 8 above)
Comment (pwilkins) [inline docs/release_process.rst]: Agreed.

File: docs/release_process.rst [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    git pull
+    git flow release start 0.10.0
+    vi lore/settings.py # change version number
+    git-release-notes v0.9.0..master util/release_notes_rst.ejs
+    vi RELEASE.rst # paste results of prior command at top of file, correcting the release number
+    git add RELEASE.rst lore/settings.py
+    git commit -m "Release 0.10.0"
+    # verify that the Travis-ci build succeeds
+    # Wait for all checkboxes to be checked
+    # Check with Dev-Ops buddy whether migrations need to be run for this release
```
Comment (pdpinch) [inline docs/release_process.rst:40]: @bdero is this step still necessary?

File: docs/release_process.rst [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+    git fetch
+    git checkout master
+    git pull
+    git flow release start 0.10.0
+    vi lore/settings.py # change version number
+    git-release-notes v0.9.0..master util/release_notes_rst.ejs
+    vi RELEASE.rst # paste results of prior command at top of file, correcting the release number
+    git add RELEASE.rst lore/settings.py
+    git commit -m "Release 0.10.0"
+    # verify that the Travis-ci build succeeds
```
Comment (pdpinch) [inline docs/release_process.rst]: There are some steps missing here about pushing the rc branch and merging it into the release-candidate branch

File: docs/release_process.rst [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+the release.  In the example given it is all commits from the last release
+to what is in ``master``.
+
+The second command argument specifies a template to format the output.
+LORE has two templates.
+
+- ``util/release_notes_rst.ejs`` formats the output as
+  reStructuredText for inclusion in ``RELEASE.rst``.
+
+- ``util/release_notes.ejs`` formats the output as to precede each commit
```
Comment (pdpinch) [inline docs/release_process.rst]: I think this is clearer:

"... formats the output as markdown so as to precede... "

File: docs/release_process.rst [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+  - one to ``release`` called ``rc/0.10.0`` containing a checklist of commits
+  - one to ``master`` called ``Release 0.10.0``
+
+Tell developers to test their changes on the LORE release candidate
+server, http://lore-rc.herokuapp.com
+and then bug team members until all the check boxes are checked.
+
+6. Hand Off Release to Dev-Ops
+------------------------------
+Lift the freeze after running migrations and then merging both the
```
Comment (pdpinch) [inline docs/release_process.rst]: no need to freeze anymore, now that we have a separate Heroku app for running release candidates

File: docs/release_process.rst [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+
+Release Process Roles
+=====================
+
+the important piece for the dev-ops person is migrations
+so the devops person should actually do the merge to release so they can run
+migrations before they do so which means we are almost exactly following the
+edX release strategy, where there is a release master and "dev-ops buddy".
+
+Peter's notes
```
Comment (pdpinch) [inline docs/release_process.rst]: I think this section is redundant now.

File: docs/release_process.rst [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+the release.
+
+.. code-block:: bash
+
+    git add RELEASE.rst lore/settings.py
+    git commit -m "Release 0.10.0"
+    git flow release publish 0.10.0
+
+The commit will trigger a build on Travis CI.
+https://travis-ci.org/mitodl/lore/builds  For each commit throughout
```
Comment (pdpinch) [inline docs/release_process.rst:90]: formatting is a little weird here. Do you want to put a link on Travis CI?

File: docs/release_process.rst [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+
+Run ``git-release-notes`` again to create the Pull Request descriptions for
+the ``release`` branch.
+
+.. code-block:: bash
+
+  git-release-notes v0.9.0..master util/release_notes.ejs
+
+and paste the Markdown output as the description of the PR to the release
+branch ``rc/0.10.0`` (This is the one with the checkboxes.)
```
Comment (pdpinch) [inline docs/release_process.rst:118]: Do you want to move these lines (106-107) after the creation of the PRs (112-115)?

File: docs/release_process.rst (hunk 11 above)
Comment (pwilkins) [inline docs/release_process.rst]: I like it!

File: docs/release_process.rst [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+Make three PRs based on the release branch you just published:
+
+  - One to the ``release-candidate`` branch named ``rc/0.10.0``
+  - Merge this PR immediately to kick off the build.
+  - One to the ``release`` branch named ``rc/0.10.0`` containing a checklist of commits
+  - one to the ``master`` branch named ``Release 0.10.0``
+
+Run ``git-release-notes`` with the other template to create the
+Pull Request descriptions for the ``release`` branch.
+
```
Comment (pdpinch) [inline docs/release_process.rst]: I still think you want to create the PR descriptions before opening the PR, i.e. put lines 110-11 before 103.

**Comment by pdpinch** [comment]
You can merge this, but I still think you should move the lines I mentioned in https://github.com/mitodl/lore/pull/636#discussion_r40423834

File: docs/release_process.rst (hunk 15 above)
Comment (pwilkins) [inline docs/release_process.rst:118]: I will move those lines before I merge.

**Pull request merged**